python/triton/compiler/compiler.py

```diff
@@ -234,7 +234,11 @@ def convert_type_repr(x):
     return x
 
 
-def make_hash(fn, target, env_vars, **kwargs):
+def make_hash(fn, target, env_vars, device_backend, **kwargs):
+    if device_backend is None:
+        version_key = get_cuda_version_key()
```

Comment:
I'm wondering if the problem comes from here, this will hash the triton library everytime which is not cheap.